Watch recognize crop

gkochera opened the pull request:
To test:
Ideally, you should setup a virtual environment before running this:
`python3 -m venv venv`
`source ./venv/bin/activate`

Install dependencies by doing
`pip3 install -r requirements.txt`

Then run by doing
`python3 carzam.py`

Then while Carzam is running, add a picture with a car or cars in it to ./data/testfiles

Carzam will automatically process the input file and output the cropped images to ./data/outfiles

I would like to get this merged to main. It's a significant step forward for us and we can begin integrating at least this part of it with the web application that Kevin is working on. Any input to my code is appreciated!

gkochera committed:
got some basic working code that implements PyTorch. working with some small datasets from autotrader currently, accuracy is low

gkochera committed:
expanded header to cover source of code; pulled out some hypertuning variables to the top like batch-size; started testing against some basic cars

gkochera committed:
added a function to test against all vehicles in the verify folder

gkochera committed:
good finish point for the first night, need to do more research on tensors and neural networks in genreal

gkochera committed:
working with yolov3, trying to keep this pure python, was able to get bounding boxes, need to work more with the library to fully understand

gkochera committed:
able to return coordinates of bounding boxes now

gkochera committed:
got a basic workflow going using PyTorch Hub for YOLOv5 and Python Image Library

gkochera committed:
test images - single car

gkochera committed:
reliably grabbing a group of files and batch processing them now

gkochera committed:
script sucessfully ids and crops photos with more than 1 car now

gkochera committed:
updated outfile data

gkochera committed:
commented code and fixed file endings for outfiles

gkochera committed:
generated a requirements.txt to ease sharing of project dependencies

gkochera committed:
deleted outfiles

gkochera committed:
prevented git from capturing outfiles

gkochera committed:
outfiles are named correctly now

gkochera committed:
added header block

gkochera committed:
refactored project into multiple files to stay organized and containerize the different elements

gkochera committed:
successfully got polling working, project now watches a directory, and outputs the cropped files to an output directory

gkochera committed:
added header

gkochera committed:
formatting

gkochera committed:
pruned extra files

gkochera committed:
removed cached weights

gkochera committed:
removed unnecessary folders

joykm reviewed (commented):
Curious if the "outimages" folder is missing or if my local run had issues generating it, otherwise looks good.

gkochera committed:
added docstrings and linted project, requirements.txt is up to date and the observer now will wait for a file to be fully written before it attempts to ID objects in it

gkochera committed:
tested on image with 10 cars, revised the docstring to discuss MIN_SIZE

gkochera commented:
- Fixed issue with missing outimages and testimages folders. If they don't exist, they will be created.
- Fixed issue with images smaller than a certain size being processed. See docstring in cropper.py for details
- Fixed issue with slowly uploading files causing Carzam to break (it would try to process a partial image file). Implemented size polling.

muse-gabs commented:
downloaded and ran the image cropper and it worked with no issues! Looks good to be merged!
